docs/example-react-router.mdx

```diff
@@ -3,7 +3,159 @@ id: example-react-router
+        <input name="person" placeholder="Name" />
+        <button type="submit">Submit</button>
+      </Form>
+    </>
+  );
+};
+
+export const Home = () => {
+  let data = useLoaderData();
+  return <div>You are {data.message}</div>;
+};
+
+export const NoMatch = () => <div>No match</div>;
+
+export const LocationDisplay = () => {
+  const location = useLocation();
+
+  return <div data-testid="location-display">{location.pathname}</div>;
+};
+
+export const Layout = () => (
+  <div>
+    <Link to="/">Home</Link>
+    <Link to="/about">About</Link>
+    <Outlet />
+    <LocationDisplay />
+  </div>
+);
+
+export const routes = [
+  {
+    path: '/',
+    element: <Layout />,
+    children: [
+      {
+        index: true,
+        element: <Home />,
+        loader: homeLoader,
+      },
+      {
+        path: '/about',
+        element: <About />,
+        action: aboutAction,
+      },
+      {
+        path: '*',
+        element: <NoMatch />,
+      },
+    ],
+  },
+];
+
+const router = createBrowserRouter(routes);
+
+const App = () => <RouterProvider router={router}></RouterProvider>;
+
+export default App;
+```
+
+```jsx
+// App.test.jsx
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import '@testing-library/jest-dom';
+import {
+  createBrowserRouter,
+  createMemoryRouter,
+  RouterProvider,
+} from 'react-router-dom';
+import { routes } from './App';
+
+test('full app rendering/navigating', async () => {
+  const router = createBrowserRouter(routes);
+  render(<RouterProvider router={router}></RouterProvider>);
+
+  const user = userEvent.setup();
+  // We need to wait for the loader data and then assert presence
+  expect(await screen.findByText(/you are home/i)).toBeInTheDocument();
+
+  // verify page content for expected route after navigating
+  await user.click(screen.getByText(/about/i));
+  expect(screen.getByText(/you are on the about page/i)).toBeInTheDocument();
+});
+
+test('landing on a bad page', () => {
+  const badRoute = '/some/bad/route';
+  const router = createMemoryRouter(routes, { initialEntries: [badRoute] });
+
+  // use createMemoryRouter when you want to manually control the history
+  render(<RouterProvider router={router}></RouterProvider>);
+
+  // verify navigation to "no match" route
+  expect(screen.getByText(/no match/i)).toBeInTheDocument();
+});
+
+test('rendering a component that uses useLocation', () => {
+  const route = '/some-route';
+  const router = createMemoryRouter(routes, { initialEntries: [route] });
+
+  // use createMemoryRouter when you want to manually control the history
+  render(<RouterProvider router={router}></RouterProvider>);
+
+  // verify location display is rendered
+  expect(screen.getByTestId('location-display')).toHaveTextContent(route);
+});
+```
+Refer to [this working example](https://stackblitz.com/edit/vitejs-vite-dnutcg?file=src%2FApp.test.jsx,src%2FApp.jsx)
```

Comment:
Because other examples don't have a stackblitz I was thinking on removing this?

Reply:
Yes, please... the reason for having that was, I couldn't get other examples to run without making code changes